### PR TITLE
KOGITO-8455: added Release Notes creation methods

### PR DESCRIPTION
**Pull request by mbiarnes**
JIRA: [KOGITO-8455](https://issues.redhat.com/browse/KOGITO-8455)

Dep PRs: 
- [kogito-runtimes](https://github.com/kiegroup/kogito-runtimes/pull/3017)
- [kogito-apps](https://github.com/kiegroup/kogito-apps/pull/1740)
- [kogito-examples](https://github.com/kiegroup/kogito-examples/pull/1678)
- [jenkins-pipeline-shared-libraries](https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/275)

### CODE REVIEW

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
please also update the title to a more meaningful one

**Comment by mbiarnes** [comment]
@radtriste a test was added

File: test/vars/GithubScmSpec.groovy [hunk 1]
```diff
@@ -1144,4 +1144,36 @@ class GithubScmSpec extends JenkinsPipelineSpecification {
...
+        groovyScript.getBinding().setVariable("GH_TOKEN", 'password')
+        when:
+        groovyScript.updateReleaseBody('tag')
+        then:
+        1 * getPipelineMock("usernamePassword.call").call(['credentialsId': 'kie-ci', 'usernameVariable': 'GH_USER', 'passwordVariable': 'GH_TOKEN']) >> 'userNamePassword'
+        1 * getPipelineMock("withCredentials")(['userNamePassword'], _ as Closure)
+        1 * getPipelineMock("sh")('gh release edit tag -F release_notes')
+    }
+
+    def "[githubscm.groovy] getPreviousTag"() {
```
Comment (radtriste) [inline test/vars/GithubScmSpec.groovy:1172]: please also test the case when latestTag == tag you send as parameter

**Comment by mbiarnes** [comment]
@radtriste done - tests are added. This PR (and related ones) should be merged before the next release to have the release creation automatically

**Review by radtriste — APPROVED** [review]

**Pull request merged**